src/main/kotlin/app/revanced/patcher/util/Document.kt

```diff
@@ -23,6 +24,22 @@ class Document internal constructor(
         readerCount.merge(file, 1, Int::plus)
     }
 
+    private fun fixSurrogatePairs(xml: String): String {
+        val regex = """&#(\d+);&#(\d+);""".toRegex()
```

Comment:
This looks like a hacky attempt to fix the issue. Isn't it an encoding issue? Perhaps the correct encoding can be specified for the writer instead?